xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -282,8 +285,14 @@ public String getQuery(XWikiContext context)
                     select.append("doc." + idField);
                 } else {
                     select.append("idprop.value");
-                    fromStatements.add("StringProperty as idprop");
                     whereStatements.add("obj.id=idprop.id.id and idprop.id.name='" + idField + "'");
+                    // Get the from statements according to the type of property
+                    try {
+                        String classType = getPropertyType(classname, idField,context);
```

Comment:
There's a missing space before ``context``. Have you formatted your code? You need to follow the [Java Code Style](http://dev.xwiki.org/xwiki/bin/view/Community/JavaCodeStyle).